### PR TITLE
Implement internal math module

### PR DESCRIPTION
**Pull request by samuelselleck**
This implements an internal math module, to be exposed to users of pax, and removes the use of kurbo::Affine as the engines main transform type.

### CODE REVIEW

File: pax-engine/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@ pub extern crate pax_macro;
 pub use pax_macro::*;
 
 pub use pax_runtime::api;
+pub use pax_runtime::design_utils;
```
Comment (zackbrown) [inline pax-engine/src/lib.rs:5]: please remove

File: pax-runtime/src/engine/design_utils.rs
Comment (zackbrown) [inline pax-runtime/src/engine/design_utils.rs:1]: please remove

File: pax-runtime/src/engine/mod.rs [hunk 2]
```diff
@@ -8,17 +8,20 @@ use std::rc::Rc;
...
 use piet::InterpolationMode;
 
 use crate::declarative_macros::{handle_vtable_update, handle_vtable_update_optional};
 use crate::{
-    Affine, ComponentInstance, ExpressionContext, InstanceNode, RuntimeContext,
+    ComponentInstance, ExpressionContext, InstanceNode, RuntimeContext,
     RuntimePropertiesStackFrame, TransformAndBounds,
 };
 
+pub mod design_utils;
```
Comment (zackbrown) [inline pax-runtime/src/engine/mod.rs:23]: please remove

**Review by zackbrown — CHANGES_REQUESTED** [review]
Nice improvements!  Small changes requested re: `design_utils`, per team conversation.  After that, please go ahead and merge directly.

**Comment by samuelselleck** [comment]
This is done now! 
- Created a new NodeInterface type that is a public wrapper over ExpandedNode
- Added methods on NodeContext to do raycasting, finding by id, etc (forwarding calls to render_context and wrapping results in NodeInterface)
- Renamed NodeContext to EngineContext to better reflect it's ability to do things like global raycasting

**Comment by warfaj** [comment]
LGTM! Thanks for the changes.

**Pull request merged**